utils/check_freestanding_size.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+import argparse
+import subprocess
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--path", help="path to a binary to check")
+parser.add_argument("--vendor", help="flavor of the freestanding stdlib")
+parser.add_argument("--size-path", help="path to llvm-size binary to use")
+args = parser.parse_args()
+
+
+################################################################################
+#
+# EXPECTED/ALLOWED TEXT SEGMENT SIZE
+#
+# Before bumping the maximum text segment size please consult with:
+# @kubamracek, @compnerd
+#
+# The 'freestanding' build of the Swift runtime and standard library is
+# intended to be lean and allow dead-code elimination from the standard library
+# based on client code usage.
+#
+################################################################################
+
+# As of 2021/12/20: TEXT segment is 704 kB on Darwin, stdlib with asserts.
+# Let's allow for some minor increases (~ 15%) and cap at 800 kB:
+
+vendor_apple_max_text_size = 800 * 1024
+
+################################################################################
+
+
+if args.vendor == "apple":
```

Comment:
I'm not a fan of basing this on vendors personally.  I think it would be better to just use the target.  There isn't anything particularly vendor specific to this.

Reply:
By target, do you mean the target triple, e.g. "x86_64-apple-macos" or something else here?